src/BizHawk.Client.Common/movie/import/LsmvImport.cs

```diff
@@ -14,20 +15,25 @@ namespace BizHawk.Client.Common.movie.import
 	[ImporterFor("LSNES", ".lsmv")]
 	internal class LsmvImport : MovieImporter
 	{
+		private static readonly byte[] Zipheader = { 0x50, 0x4b, 0x03, 0x04 };
 		private LibsnesControllerDeck _deck;
 		protected override void RunImport()
 		{
-			Result.Movie.HeaderEntries[HeaderKeys.Core] = CoreNames.Bsnes;
-
-			var hf = new HawkFile(SourceFile.FullName);
-
 			// .LSMV movies are .zip files containing data files.
-			if (!hf.IsArchive)
+			using (var fs = new FileStream(SourceFile.FullName, FileMode.Open, FileAccess.Read))
 			{
-				Result.Errors.Add("This is not an archive.");
-				return;
+				byte[] data = new byte[4];
+				fs.Read(data, 0, 4);
+				if (!data.SequenceEqual(Zipheader))
+				{
+					Result.Errors.Add("This is not a zip file.");
+					return;
+				}
 			}
 
+			using var zip = new ZipFile(SourceFile.FullName);
+
+			Result.Movie.HeaderEntries[HeaderKeys.Core] = CoreNames.Bsnes;
```

Comment:
Does the Core header need to be set before failure?

The rest looks good, though I wish you'd used SharpCompress.

Reply:
I've restored the original ordering of the header setting.

Regarding SharpCompress, this was not an exercise in innovation or imagination; I used the library I did because I simply copied working code from other systems.